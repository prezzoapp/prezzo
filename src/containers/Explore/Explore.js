// @flow
import React, {Component} from 'react';
import {View, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/MaterialIcons';

class Explore extends Component {
  static displayName = 'Explore';

  static navigationOptions = {
    title: 'Explore',
    tabBarIcon: props => (
        <Icon name='explore' size={24} color={props.tintColor} />
    ),
    headerTintColor: 'white',
    headerStyle: {
      position: 'absolute',
      backgroundColor: 'transparent',
      zIndex: 100,
      top: 0,
      left: 0,
      right: 0
    }
  };

  static propTypes = {
    navigate: PropTypes.func.isRequired
  };

  render() {
    return (
      <View style={styles.container} />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#2B2C2C'
  }
});

export default Explore;
