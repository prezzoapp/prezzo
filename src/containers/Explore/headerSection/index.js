// @flow
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {NavigationActions} from 'react-navigation';
import HeaderSection from './headerSection';

import {toggleFilter, selectedFilterID} from './actions';

const mapStateToProps = (state) => {
  return {
    filtersReducer: state.get('filtersReducer'),
    // selectedFilterID: state.get('selectedFilterIDReducer')
  };
};

export default connect(
  mapStateToProps,
  dispatch => {
    return {
      navigate: bindActionCreators(NavigationActions.navigate, dispatch),
      toggleFilter: bindActionCreators(toggleFilter, dispatch),
      // selectedFilterID: bindActionCreators(selectedFilterID, dispatch)
    };
  }
)(HeaderSection);
