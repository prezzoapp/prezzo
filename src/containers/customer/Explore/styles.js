// @flow
import { StyleSheet } from 'react-native';
import { FONT_FAMILY, COLOR_WHITE, SF_PRO_TEXT_SEMI_BOLD,
SF_PRO_DISPLAY_SEMI_BOLD,
SF_PRO_DISPLAY_REGULAR } from '../../../services/constants';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen';
import { getBottomSpace } from 'react-native-iphone-x-helper';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#2B2C2C'
  },

  loaderContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.8)'
  },

  message: {
    color: COLOR_WHITE,
    fontFamily: FONT_FAMILY,
    fontSize: 15,
    marginTop: 20
  },

  filtersHolder: {
    paddingVertical: wp('5.33%'),
    position: 'absolute',
    left: 0,
    right: 0,
    zIndex: 1
  },

  filtersList: {
    paddingLeft: wp('4.53%')
  },

  slidersHolder: {
    paddingHorizontal: wp('8%'),
    paddingTop: hp('1.84%')
  },

  sliderTitleText: {
    color: 'rgb(255,251,245)',
    fontSize: wp('2.93%'),
    fontFamily: SF_PRO_DISPLAY_REGULAR
  },

  sliderTitleHolder: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },

  blurView: {
    ...StyleSheet.absoluteFillObject
  },

  priceBarIndicator: {
    width: 2,
    height: 12,
    zIndex: 1,
    backgroundColor: 'white'
  },

  priceSliderContainer: {
    height: 31,
    marginVertical: hp('1.84%'),
    marginLeft: wp('8%'),
    marginRight: wp('10.66%'),
    position: 'relative',
    flexDirection: 'row',
    justifyContent: 'space-between'
  },

  priceSliderHolder: {
    height: 31,
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: -10.25
  },
  
  thumbStyle: {
    height: 13,
    width: 13
  },

  trackStyle: {
    height: 3
  },

  sliderHeight: {
    height: 31
  },

  bellBtnHolder: {
    zIndex: 1,
    height: wp('12.26%'),
    width: wp('12.26%'),
    borderRadius: wp('6.13%'),
    borderWidth: 1,
    borderColor: '#0DD24A',
    position: 'absolute',
    right: wp('6.13%'),
    bottom: getBottomSpace() + 49 + wp('3.46%'),
    justifyContent: 'center',
    alignItems: 'center'
  }
});
