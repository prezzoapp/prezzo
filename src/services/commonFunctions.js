export const getTimeStampString = () => new Date().getTime().toString();

export const findById = (tree, testID) => {
  if (tree.props && tree.props.testID === testID) {
    return tree;
  }
  if (tree.children && tree.children.length > 0) {
    let childs = tree.children;
    for (let i = 0; i < childs.length; i++) {
      let item = findById(childs[i], testID);
      if (typeof item !== 'undefined') {
        return item;
      }
    }
  }
};
