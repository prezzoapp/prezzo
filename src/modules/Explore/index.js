// @flow
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {NavigationActions} from 'react-navigation';
import Explore from './Explore';

export default connect(
  null,
  dispatch => {
    return {
      navigate: bindActionCreators(NavigationActions.navigate, dispatch)
    };
  }
)(Explore);
