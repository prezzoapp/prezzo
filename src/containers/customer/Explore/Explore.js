// @flow
import React, { PureComponent } from 'react';
import { View, ActivityIndicator, Text, Modal, Animated, FlatList, Platform, NetInfo } from 'react-native';
import { LinearGradient, BlurView, Location, Permissions } from 'expo';
import PropTypes from 'prop-types';
import { MaterialIcons } from '../../../components/VectorIcons';
import ExploreSearch from '../ExploreSearch';
import ExploreScreenHeader from '../ExploreScreenHeader';
import ExploreList from '../ExploreList';
import ExploreSearchInput from '../../../components/ExploreSearchInput';
import FilterItem from '../../../components/FilterItem';
import Slider from 'react-native-slider';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import styles from './styles';
import { get } from '../../../utils/api';
import LoadingComponent from '../../../components/LoadingComponent';
import Toast from 'react-native-custom-toast';
import {
  COLOR_GREEN,
  FONT_FAMILY_MEDIUM,
  SF_PRO_DISPLAY_REGULAR
} from '../../../services/constants';

import { showAlertWithMessage, manuallyLogout } from '../../../services/commonFunctions';

const price2Indicator = wp('85%') * 0.33 - wp('6.66%');

const price3Indicator = wp('85%') * 0.66 - wp('9.5%');

const price4Indicator = wp('85%') * 0.99 - wp('9.5%');

let disableBtn = false;

class Explore extends PureComponent<Props> {
  static navigationOptions = {
    title: 'Explore',
    tabBarIcon: props => (
      <MaterialIcons name="explore" size={24} color={props.tintColor} />
    ),
    headerTintColor: 'white',
    headerStyle: {
      position: 'absolute',
      backgroundColor: 'transparent',
      zIndex: 100,
      top: 0,
      left: 0,
      right: 0
    }
  };

  static displayName = 'Explore';

  constructor(props) {
    super(props);

    this.state = {
      showList: false,
      filteredData: [],
      showLoader: false,
      showFilters: false
    };

    this.animatedValue = new Animated.Value(0);

    this.timeOutVar = -1;

    this.activeFilters = [];
  }

  componentDidMount() {
    this.props.filters.map(item => {
      if(item.on) {
        this.activeFilters.push(item.filterType);
      }
    });

    this.hitAPI();
  }

  showToast() {
    this.toast.showToast('Location services off, got location through IP.');
  }

  hitAPI() {
    this.props.getUserCurrentLocation().then(coords => {
      this.props.listVendors(
        coords.latitude,
        coords.longitude,
        this.props.distance,
        this.activeFilters.join(','),
        this.props.pricing
      ).then(() => {
          if(coords.gotThrough && coords.gotThrough === 'IP') {
            this.showToast();
          }
          disableBtn = false;
        })
        .catch(err => {
          if(err.code === 401) {
            manuallyLogout(err, () => this.props.userLogout());
          } else {
            showAlertWithMessage('Uh-oh!', err, () => {
              disableBtn = false;
            })
          }
        });
    }).catch(err => showAlertWithMessage('Uh-oh!', err, () => {
        disableBtn = false;
    }));
  }

  onTextChange(text) {
    if(this.state.showLoader === false) {
      this.setState(() => {
        return {
          showLoader: true
        }
      });
    }
    if(text !== '') {
      this.clearTimer();
      this.timeOutVar = setTimeout(() => {
        this.callWebService(text);
      }, 2000);
    }
  }

  clearTimer() {
    clearTimeout(this.timeOutVar);
    this.timeOutVar = -1;
  }

  async callWebService(text) {
    try {
      await get(`/v1/vendors?name=${text}`).then(response => {
        this.setState(() => {
          return {
            filteredData: response,
            showLoader: false
          }
        });
      });
    } catch (err) {
        if(err.code === 401) {
          this.setState({ showLoader: false }, () => {
            manuallyLogout(err, () => this.props.userLogout());
          });
        } else {
          this.setState({ showLoader: false }, () => {
            showAlertWithMessage('Uh-oh!', err);
          });
        }
      }
  }

  showList(value) {
    this.setState(() => {
      return {
        showList: value
      }
    });
  }

  setFilterPanelPosition = (y, height) => {
    this.animatedValue.setValue(y + height);
    console.log(y, height);
  }

  showFilters = () => {
    this.setState({ showFilters: !this.state.showFilters });
  }

  updateDistance(distance) {
    const newDistance = parseFloat(distance.toFixed(1));
    this.props.updateDistance(newDistance).then(() => {
      this.hitAPI();
    });
  }

  updatePrice(price) {
    this.props.updatePrice(price).then(() => {
      this.hitAPI();
    });
  }

  toggleFilter = (id) => {
    if(disableBtn === false) {
      disableBtn = true;
      this.activeFilters = [];
      this.props.toggleFilter(id).then(() => {
        this.props.filters.map(item => {
          if(item.on) {
            this.activeFilters.push(item.filterType);
          }
        });
        this.hitAPI();
      });
    }
  }

  render() {
    const { filters } = this.props;
    return (
      <LinearGradient
        testID="linearGradient"
        colors={['rgb(0,0,0)', 'transparent', 'transparent', 'transparent', 'transparent']}
        locations={[0.1, 0.4, 0.4, 0.4, 0.4]}
        style={styles.container}
      >
        <View style={{ flex: 1 }}>
          <ExploreSearchInput
            showList={value => this.showList(value)}
            showListValue={this.state.showList}
            onTextChange={text => this.onTextChange(text)}
            clearTimer={this.clearTimer}
          />
          <ExploreScreenHeader
            testID="exploreHeader"
            setFilterPanelPosition={this.setFilterPanelPosition}
            showFilters={this.showFilters}
          />

          {/* FILTERS PANEL UI MOVED FROM EXPLORE SCREEN HEADER COMPONENT. */}
          {this.state.showFilters &&
            <Animated.View
              style={[
                  styles.filtersHolder, {
                    transform: [{ translateY: this.animatedValue }]
            }]}>
              <BlurView style={styles.blurView} tint="dark" intensity={90} />
              <FlatList
                horizontal
                contentContainerStyle={styles.filtersList}
                keyExtractor={item => item._id.toString()}
                showsHorizontalScrollIndicator={false}
                data={filters}
                renderItem={({ item }) =>
                  <FilterItem
                    image={item.image}
                    name={item.name}
                    on={item.on}
                    toggleFilter={() => this.toggleFilter(item._id)}
                  />
                }
              />

              <View>
                <View style={styles.slidersHolder}>
                  <View style={styles.sliderTitleHolder}>
                    <Text style={styles.sliderTitleText}>Distance</Text>
                    <Text style={styles.sliderTitleText}>
                      {this.props.maxDistance}mi
                    </Text>
                  </View>
                  <Slider
                    minimumValue={this.props.minDistance}
                    maximumValue={this.props.maxDistance}
                    minimumTrackTintColor="rgb(47,212,117)"
                    maximumTrackTintColor="rgb(230,230,230)"
                    thumbTintColor="rgb(255,254,255)"
                    thumbStyle={{ height: 18, width: 18 }}
                    value={this.props.distance}
                    trackStyle={{ height: 3 }}
                    thumbStyle={{ height: 13, width: 13 }}
                    onSlidingComplete={value => this.updateDistance(value)}
                  />
                </View>
                {filters.map(item => {
                  if (item.filterType === 'price' && item.on === true) {
                    return (
                      <View
                        key={item._id}
                        style={styles.priceSliderContainer}
                      >
                        <View
                          style={styles.priceSliderHolder}
                        >
                          <Slider
                            minimumValue={0}
                            maximumValue={3}
                            step={1}
                            minimumTrackTintColor="rgb(47,212,117)"
                            maximumTrackTintColor="rgb(230,230,230)"
                            thumbTintColor="rgb(255,254,255)"
                            thumbStyle={{ height: 18, width: 18 }}
                            style={{height: 31}}
                            value={this.props.pricing - 1}
                            onSlidingComplete={value => this.updatePrice(value)}
                            trackStyle={{ height: 3 }}
                            thumbStyle={{ height: 13, width: 13 }}
                          />
                        </View>

                        <View
                          pointerEvents="none"
                          style={{
                            width: wp('13.33%'),
                            height: 31,
                            flexDirection: 'column',
                            justifyContent: 'space-between',
                            zIndex: 1
                          }}
                        >
                          <Text
                            style={{
                              color: COLOR_GREEN,
                              fontSize: wp('2.66%'),
                              width: wp('13.33%'),
                              height: 20,
                              fontFamily: SF_PRO_DISPLAY_REGULAR
                            }}
                          >
                            $
                          </Text>

                          <View
                            style={[
                              styles.priceBarIndicator,
                              {
                                backgroundColor:
                                  this.props.pricing - 1 > 0.0
                                    ? 'rgba(255,255,255,1.0)'
                                    : null
                              }
                            ]}
                          />
                        </View>

                        <View
                          pointerEvents="none"
                          style={{
                            width: wp('13.33%'),
                            left: price2Indicator,
                            position: 'absolute',
                            height: 31,
                            flexDirection: 'column',
                            zIndex: 1,
                            justifyContent: 'space-between',
                            alignItems: 'center'
                          }}
                        >
                          <Text
                            style={[
                              {
                                fontSize: wp('2.66%'),
                                height: 20,
                                textAlign: 'center',
                                width: wp('13.33%'),
                                fontFamily: SF_PRO_DISPLAY_REGULAR
                              },
                              {
                                color:
                                  this.props.pricing - 1 >= 1
                                    ? COLOR_GREEN
                                    : 'rgba(255,255,255,1.0)'
                              }
                            ]}
                          >
                            $$
                          </Text>

                          <View
                            style={[
                              styles.priceBarIndicator,
                              {
                                backgroundColor:
                                  this.props.pricing - 1 === 1
                                    ? null
                                    : 'rgba(255,255,255,1.0)'
                              }
                            ]}
                          />
                        </View>

                        <View
                          pointerEvents="none"
                          style={{
                            width: wp('13.33%'),
                            position: 'absolute',
                            left: price3Indicator,
                            height: 31,
                            flexDirection: 'column',
                            zIndex: 1,
                            justifyContent: 'space-between',
                            alignItems: 'center'
                          }}
                        >
                          <Text
                            style={[
                              {
                                fontSize: wp('2.66%'),
                                height: 20,
                                marginLeft: 2,
                                textAlign: 'center',
                                width: wp('13.33%'),
                                fontFamily: SF_PRO_DISPLAY_REGULAR
                              },
                              {
                                color:
                                  this.props.pricing - 1 >= 2
                                    ? COLOR_GREEN
                                    : 'rgba(255,255,255,1.0)'
                              }
                            ]}
                          >
                            $$$
                          </Text>

                          <View
                            style={[
                              styles.priceBarIndicator,
                              {
                                backgroundColor:
                                  this.props.pricing - 1 === 2
                                    ? null
                                    : 'rgba(255,255,255,1.0)'
                              }
                            ]}
                          />
                        </View>

                        <View
                          pointerEvents="none"
                          style={{
                            width: wp('13.33%'),
                            height: 31,
                            position: 'absolute',
                            left: price4Indicator,
                            flexDirection: 'column',
                            zIndex: 1,
                            justifyContent: 'space-between',
                            alignItems: 'center'
                          }}
                        >
                          <Text
                            style={[
                              {
                                fontSize: wp('2.66%'),
                                height: 20,
                                textAlign: 'center',
                                width: wp('13.33%'),
                                fontFamily: SF_PRO_DISPLAY_REGULAR
                              },
                              {
                                color:
                                  this.props.pricing - 1 >= 3
                                    ? COLOR_GREEN
                                    : 'rgba(255,255,255,1.0)'
                              }
                            ]}
                          >
                            $$$$
                          </Text>

                          <View
                            style={[
                              styles.priceBarIndicator,
                              {
                                backgroundColor:
                                  this.props.pricing - 1 >= 3
                                    ? null
                                    : 'rgba(255,255,255,1.0)'
                              }
                            ]}
                          />
                        </View>
                      </View>
                    );
                  }
                })}
              </View>
            </Animated.View>
          }

          <ExploreList
            testID="exploreList"
            showToast={() => this.showToast()}
          />
          {this.state.showList &&
            <ExploreSearch
              testID="exploreSearch"
              showLoader={this.state.showLoader}
              filteredData={this.state.filteredData}
            />
          }
        </View>
        <LoadingComponent visible={this.props.isBusy} />
        <Toast
          ref={toast => this.toast = toast}
          position='bottom'
          orientation='yAxis'
          backgroundColor={COLOR_GREEN}
        />
      </LinearGradient>
    );
  }
}

Explore.propTypes = {
  filters: PropTypes.array.isRequired,
  getUserCurrentLocation: PropTypes.func.isRequired,
  updateDistance: PropTypes.func.isRequired,
  updatePrice: PropTypes.func.isRequired,
  toggleFilter: PropTypes.func.isRequired,
  listVendors: PropTypes.func.isRequired,
  userLogout: PropTypes.func.isRequired,
  distance: PropTypes.number.isRequired,
  pricing: PropTypes.number.isRequired,
  minDistance: PropTypes.number.isRequired,
  maxDistance: PropTypes.number.isRequired
};

export default Explore;
