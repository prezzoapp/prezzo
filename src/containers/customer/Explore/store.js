// @flow
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';
import { listVendors, toggleFilter, updateDistance, updatePrice, getUserCurrentLocation, callWaiterBtnFunc } from '../../../modules/explore';

export const mapStateToProps = state => {
  const filters = state
    .get('explore')
    .get('filters')
    .toJS();
  const distance = state.get('explore').get('distance');
  const pricing = state.get('explore').get('pricing');
  const isBusy = state.get('explore').get('isBusy');
  const minDistance = state.get('explore').get('minDistance');
  const maxDistance = state.get('explore').get('maxDistance');
  const callWaiterBtnState = state.get('explore').get('callWaiterBtnState')
  const userName = state.get('user').get('account').get('fullName');
  const isUserOpenOrder = state.get('userActivity').get('data').size !== 0 ? true : false

  return {
    filters,
    distance,
    pricing,
    isBusy,
    minDistance,
    maxDistance,
    userName,
    callWaiterBtnState,
    isUserOpenOrder
  };
};

export const mapDispatchToProps = dispatch => ({
  navigate: bindActionCreators(NavigationActions.navigate, dispatch),
  listVendors: bindActionCreators(listVendors, dispatch),
  toggleFilter: bindActionCreators(toggleFilter, dispatch),
  updateDistance: bindActionCreators(updateDistance, dispatch),
  updatePrice: bindActionCreators(updatePrice, dispatch),
  getUserCurrentLocation: bindActionCreators(getUserCurrentLocation, dispatch),
  callWaiterBtnFunc: bindActionCreators(callWaiterBtnFunc, dispatch)
});
