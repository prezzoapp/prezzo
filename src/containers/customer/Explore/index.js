// @flow
import { connect } from 'react-redux';
import Explore from './Explore';
import { mapStateToProps, mapDispatchToProps } from './store';

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Explore);
