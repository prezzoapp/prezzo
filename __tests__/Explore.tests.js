import 'react-native';
import React from 'react';
import Explore from '../src/containers/customer/Explore/Explore';
import Enzyme from 'enzyme';
import { findById } from '../src/services/commonFunctions';
import renderer from 'react-test-renderer';

it('Tourial Screen renders correctly', () => {
  const explore = renderer.create(<Explore />).toJSON();
  expect(explore).toMatchSnapshot();
});

it('it should render a linearGradient', () => {
  const explore = renderer.create(<Explore />).toJSON();
  expect(findById(explore, 'linearGradient')).toBeDefined();
});

it('it should render a exploreList', () => {
  const wrapper = Enzyme.shallow(<Explore />);
  expect(wrapper.find('ExploreList').at(0));
});

it('it should render a loadingText', () => {
  const explore = renderer.create(<Explore />).toJSON();
  expect(findById(explore, 'loadingText')).toBeDefined();
});

it('it should render a activityIndicator', () => {
  const explore = renderer.create(<Explore />).toJSON();
  expect(findById(explore, 'activityIndicator')).toBeDefined();
});

it('Check State for Fetching', () => {
  const wrapper = Enzyme.shallow(<Explore />);
  expect(wrapper.instance().state.isFetching).toEqual(true);
});
