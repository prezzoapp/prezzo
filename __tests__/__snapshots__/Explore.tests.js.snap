// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tourial Screen renders correctly 1`] = `
<ExponentLinearGradient
  colors={
    Array [
      4278190080,
      0,
      0,
    ]
  }
  endPoint={undefined}
  locations={
    Array [
      0,
      0.5,
      0.5,
    ]
  }
  startPoint={undefined}
  style={
    Object {
      "backgroundColor": "#2B2C2C",
      "flex": 1,
    }
  }
  testID="linearGradient"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="white"
      hidesWhenStopped={true}
      size="large"
      testID="activityIndicator"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "System",
          "fontSize": 15,
          "marginTop": 20,
        }
      }
      testID="loadingText"
    >
      Please wait, While fetching restaurants.
    </Text>
  </View>
</ExponentLinearGradient>
`;
